### PR TITLE
Rework some code, follow the "Use Async All the Way" pattern

### PR DESCRIPTION
**Pull request by HugoX104447**
* Use async Main function, since there is no return value anyway.
* Use availabe async methods if possible.
* Do not synchronously block the thread in main.

### CODE REVIEW

File: HomeAutomationServer/Program.cs [hunk 1]
```diff
@@ -83,11 +83,11 @@ private static void Main(string[] args)
                 return;
         }
 
-        RunServicesAsync().GetAwaiter().GetResult();
+        await RunServicesAsync().ConfigureAwait(false);
 
         while (true)
         {
-            Thread.Sleep(5000);
+            await Task.Delay(5000).ConfigureAwait(false);
```
Comment (christoh) [inline HomeAutomationServer/Program.cs]:
```suggestion
            await Task.Delay(5000).ConfigureAwait(true);
```
As the while loop runs forever, it should not do so on a thread pool thread

File: HomeAutomationServer/Program.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ private static void Main(string[] args)
 
         try
         {
-            settings = Settings.Load();
+            settings = await Settings.LoadAsync().ConfigureAwait(false);
```
Comment (christoh) [inline HomeAutomationServer/Program.cs]:
```suggestion
            settings = await Settings.LoadAsync().ConfigureAwait(true);
```

File: HomeAutomationServer/Program.cs [hunk 3]
```diff
@@ -43,7 +43,7 @@ private static void Main(string[] args)
             });
             
             settings.ModbusMappings.Add(new ModbusMapping());
-            settings.Save();
+            await settings.SaveAsync().ConfigureAwait(false);
```
Comment (christoh) [inline HomeAutomationServer/Program.cs]:
```suggestion
            await settings.SaveAsync().ConfigureAwait(true);
```

File: HomeAutomationServer/Program.cs [hunk 4]
```diff
@@ -83,11 +83,11 @@ private static void Main(string[] args)
                 return;
         }
 
-        RunServicesAsync().GetAwaiter().GetResult();
+        await RunServicesAsync().ConfigureAwait(false);
```
Comment (christoh) [inline HomeAutomationServer/Program.cs]:
```suggestion
        await RunServicesAsync().ConfigureAwait(true);
```

**Comment by HugoX104447** [comment]
Hi Christoph, many thanks for taking the time to review my patches. I've double checked the documentation and reread the article quoted everywhere [ConfigureAwait FAQ](https://devblogs.microsoft.com/dotnet/configureawait-faq/).

From what I gathered I agree that in most cases it's a good practice to use `ConfigureAwait(true)` or not at all (cause it's the default behavior) in application code for asynchronous operations.  But since command-line apps "lack" a synchronization context (unlike App-level code, such as Windows Forms, WPF, and ASP.NET) using `ConfigureAwait(false)` should be the save thing to do. But here comes the caveat: I don't have an extensive background in C# programming (maybe around 2 years or so, I have over 20 years in C++ though), so I will follow all the advice given, as I can't claim to know better. 😃

**Pull request merged**